include/gatekeeper_acl.h

```diff
@@ -24,17 +24,16 @@
 #include "gatekeeper_net.h"
 
 struct acl_search {
-	/* References to the start of the IP header in each packet. */
-	const uint8_t   *data[GATEKEEPER_MAX_PKT_BURST];
-	/* References to each packet's mbuf. */
-	struct rte_mbuf *mbufs[GATEKEEPER_MAX_PKT_BURST];
-	/* The classification results for each packet. */
-	uint32_t        res[GATEKEEPER_MAX_PKT_BURST];
 	/* The number of packets held for classification. */
 	unsigned int    num;
-};
-
-#define ACL_SEARCH_DEF(name) struct acl_search name = { .num = 0, }
+	/* Fixed field here. */
+	const uint8_t   **data;
+	/* List of references to each packet's mbuf. */
+	struct rte_mbuf *mbufs[0];
+} __attribute__((packed));
```

Comment:
Why did you add `__attribute__((packed))` to `struct acl_search`? I think we shouldn't use it here.

Reply:
The same issue goes for `struct cps_bgp_req`, `struct lls_arp_req`, and `struct lls_nd_req`.